### PR TITLE
Add Olly JF PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Olly JF PCB. 
Duck Jetfire compatible.
Thank you.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/olly/jf/config.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// /*== or choose animations ==*/
+//     #define RGBLIGHT_EFFECT_BREATHING
+//     #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//     #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//     #define RGBLIGHT_EFFECT_SNAKE
+//     #define RGBLIGHT_EFFECT_KNIGHT
+//     #define RGBLIGHT_EFFECT_CHRISTMAS
+//     #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//     #define RGBLIGHT_EFFECT_RGB_TEST
+//     #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/config.h:72]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/mechlovin/olly/jf/jf.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  // if on layer 5, turn on C6 LED, otherwise off.
+    if (biton32(state) == 5) {
+        writePinHigh(C6);
+    } else {
+        writePinLow(C6);
+    }
+
+    return state;
+}
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/jf.c:66]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(D1, layer_state_cmp(state, 1));
    writePin(D0, layer_state_cmp(state, 2));
    writePin(C1, layer_state_cmp(state, 3));
    writePin(C0, layer_state_cmp(state, 4));
    writePin(C6, layer_state_cmp(state, 5));

    return state;
}
```

Alternatively, if you only want the highest layer.
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(D1, get_highest_layer(state) == 1);
    writePin(D0, get_highest_layer(state) == 2);
    writePin(C1, get_highest_layer(state) == 3);
    writePin(C0, get_highest_layer(state) == 4);
    writePin(C6, get_highest_layer(state) == 5);

    return state;
}
```

File: keyboards/mechlovin/olly/jf/keymaps/via/config.h [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 1607
```
Comment (drashna) [inline keyboards/mechlovin/olly/jf/keymaps/via/config.h:1]: This needs a license header, and an include guard (`#pragma once`)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/keymaps/via/config.h (hunk 3 above)
Comment (mechlovin) [reply keyboards/mechlovin/olly/jf/keymaps/via/config.h:1]: yes, thank you.

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/olly/jf/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_INS,  KC_HOME, KC_PGUP, KC_PSCR,
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/keymaps/default/keymap.c:20]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_INS,  KC_HOME, KC_PGUP, KC_PSCR,
```

File: keyboards/mechlovin/olly/jf/info.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
+{
+    "keyboard_name": "Mechlovin Olly",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "width": 20.5,
+    "height": 6.75,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_INS,  KC_HOME, KC_PGUP, KC_PSCR,
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/keymaps/via/keymap.c:20]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,                    KC_F13,           KC_INS,  KC_HOME, KC_PGUP, KC_PSCR,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/keymaps/via/config.h [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 1607
+#define DYNAMIC_KEYMAP_LAYER_COUNT 6
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/keymaps/via/config.h:21]:
```suggestion
```

File: keyboards/mechlovin/olly/jf/info.json [hunk 8]
```diff
@@ -0,0 +1,121 @@
+{
+    "keyboard_name": "Mechlovin Olly",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "width": 20.5,
+    "height": 6.75,
```
Comment (fauxpark) [inline keyboards/mechlovin/olly/jf/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/olly/jf/keymaps/via/config.h (hunk 7 above)
Comment (mechlovin) [reply keyboards/mechlovin/olly/jf/keymaps/via/config.h:21]: The Olly JF is made compatible with the Duck Jetfire, which has 5 layer-indicating LEDs, so I think 6 layers in VIA is suitable. Please let me know if that's ok. Thank you.

Comment (fauxpark) [reply keyboards/mechlovin/olly/jf/keymaps/via/config.h:21]: Then the keymap needs to match.

Comment (mechlovin) [reply keyboards/mechlovin/olly/jf/keymaps/via/config.h:21]: Added. Thank you.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**